### PR TITLE
Configuration file for controlling display of bulk-annotations

### PR DESCRIPTION
**Pull request by manics**
Not ready for review yet. This is on top of https://github.com/openmicroscopy/openmicroscopy/pull/4037
I'm opening this so it can hopefully get into the build for initial testing

This adds a new `PyYAML` Python requirement.

--on-hold

### CODE REVIEW

File: components/tools/OmeroPy/src/omero/util/metadata_utils.py [hunk 1]
```diff
@@ -0,0 +1,288 @@
...
+from collections import deque
+import re
+
+
+class BulkAnnotationConfiguration(object):
+    """
+    Parent class for handling bulk-annotation column configurations
+    """
+
+    REQUIRED = {"name"}
```
Comment (joshmoore) [inline components/tools/OmeroPy/src/omero/util/metadata_utils.py]:
```
(venv)[jamoore@ome-c6220-1-2 omero]$ ./OMERO-CURRENT/bin/omero metadata -h
Error loading: /opt/omero/OMERO-CURRENT/lib/python/omero/plugins/metadata.py
Traceback (most recent call last):
  File "/opt/omero/OMERO-CURRENT/lib/python/omero/cli.py", line 1367, in loadpath
    execfile(str(pathobj), loc)
  File "/opt/omero/OMERO-CURRENT/lib/python/omero/plugins/metadata.py", line 21, in <module>
    from omero.util import populate_metadata, populate_roi
  File "/opt/omero/OMERO-CURRENT/lib/python/omero/util/populate_metadata.py", line 46, in <module>
    from omero.util.metadata_utils import KeyValueListPassThrough
  File "/opt/omero/OMERO-CURRENT/lib/python/omero/util/metadata_utils.py", line 38
     REQUIRED = {"name"}
                       ^
 SyntaxError: invalid syntax
```

excluding

Comment (manics) [inline components/tools/OmeroPy/src/omero/util/metadata_utils.py]: That'll be python 2.6...

**Comment by manics** [comment]
Should be Python 2.6 compliant now

**Comment by manics** [comment]
@joshmoore should I make `import yaml` optional? Or keep it mandatory since its an easy dependency to install?

**Comment by joshmoore** [comment]
Re-included. Considering the impact (and until we are sure we always have a `virtualenv` or `conda` env), I'd think having just the commands be disabled would be preferable.

**Comment by manics** [comment]
rebased & force pushed

**Comment by manics** [comment]
Closing, see #4240 instead

**Pull request closed without merging**